core/server/src/main/java/alluxio/master/AlluxioMaster.java

```diff
@@ -186,358 +74,49 @@ public static void main(String[] args) {
-    // set up multiplexed thrift processors
-    TMultiplexedProcessor processor = new TMultiplexedProcessor();
-    registerServices(processor, mBlockMaster.getServices());
-    registerServices(processor, mFileSystemMaster.getServices());
-    if (LineageUtils.isLineageEnabled()) {
-      registerServices(processor, mLineageMaster.getServices());
-    }
-    // register additional masters for RPC service
-    for (Master master : mAdditionalMasters) {
-      registerServices(processor, master.getServices());
-    }
-
-    // Return a TTransportFactory based on the authentication type
-    TTransportFactory transportFactory;
-    try {
-      transportFactory = mTransportProvider.getServerTransportFactory();
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
-    }
-
-    // create master thrift service with the multiplexed processor.
-    Args args = new TThreadPoolServer.Args(mTServerSocket).maxWorkerThreads(mMaxWorkerThreads)
-        .minWorkerThreads(mMinWorkerThreads).processor(processor).transportFactory(transportFactory)
-        .protocolFactory(new TBinaryProtocol.Factory(true, true));
-    if (Configuration.getBoolean(PropertyKey.TEST_MODE)) {
-      args.stopTimeoutVal = 0;
-    } else {
-      args.stopTimeoutVal = Constants.THRIFT_STOP_TIMEOUT_SECONDS;
-    }
-    mMasterServiceServer = new TThreadPoolServer(args);
-
-    // start thrift rpc server
-    mIsServing = true;
-    mStartTimeMs = System.currentTimeMillis();
-    mMasterServiceServer.serve();
-  }
-
-  protected void stopServing() throws Exception {
-    if (mMasterServiceServer != null) {
-      mMasterServiceServer.stop();
-      mMasterServiceServer = null;
-    }
-    if (mWebServer != null) {
-      mWebServer.stop();
-      mWebServer = null;
-    }
-    MetricsSystem.stopSinks();
-    mIsServing = false;
-  }
-
-  /**
-   * Checks to see if the journal directory is formatted.
-   *
-   * @param journalDirectory the journal directory to check
-   * @return true if the journal directory was formatted previously, false otherwise
-   * @throws IOException if an I/O error occurs
-   */
-  private boolean isJournalFormatted(String journalDirectory) throws IOException {
-    UnderFileSystem ufs = UnderFileSystem.get(journalDirectory);
-    if (!ufs.providesStorage()) {
-      // TODO(gene): Should the journal really be allowed on a ufs without storage?
-      // This ufs doesn't provide storage. Allow the master to use this ufs for the journal.
-      LOG.info("Journal directory doesn't provide storage: {}", journalDirectory);
-      return true;
-    }
-    String[] files = ufs.list(journalDirectory);
-    if (files == null) {
-      return false;
-    }
-    // Search for the format file.
-    String formatFilePrefix = Configuration.get(PropertyKey.MASTER_FORMAT_FILE_PREFIX);
-    for (String file : files) {
-      if (file.startsWith(formatFilePrefix)) {
-        return true;
-      }
+      System.exit(-1);
```

Comment:
Is it possible to abstract this block of logic (repeats in the different services)?